File: Cpp/1700_number_of_students_unable_to_eat_lunch.cpp
// https://leetcode.cn/problems/number-of-students-unable-to-eat-lunch/

class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int eat0 = 0, eat1 = 0;
        for (int i: students) {
            if (i == 0) {
                eat0 += 1;
            }
            else {
                eat1 += 1;
            }
        }
        for (int sand: sandwiches) {
            if ((sand == 0) && (eat0 > 0)) {
                eat0 -= 1;
            }
            else if ((sand == 1) && (eat1 > 0)) {
                eat1 -= 1;
            }
            else {
                break;
            }
        }
        return eat0 + eat1;
    }
};

File: Python3/1700_number_of_students_unable_to_eat_lunch.py
# https://leetcode.cn/problems/number-of-students-unable-to-eat-lunch/

class Solution:
    def countStudents(self, students, sandwiches):
        """
        Inputs:
            students: list[int]
            sandwiches: list[int]
        Outputs:
            res: int
        """

        eat1 = sum(students)
        eat0 = len(students) - eat1

        for sand in sandwiches:
            if sand == 0 and eat0 > 0:
                eat0 -= 1
            elif sand == 1 and eat1 > 0:
                eat1 -= 1
            else:
                break
        return eat0 + eat1
